framework/src/modules/dpos_v2/stores/eligible_delegates.ts

```diff
@@ -30,20 +33,72 @@ export const eligibleDelegatesStoreSchema = {
 	},
 };
 
+// uint64 + address
+const KEY_LENGTH = 8 + 20;
+
 export class EligibleDelegatesStore extends BaseStore<EligibleDelegate> {
 	public schema = eligibleDelegatesStoreSchema;
 
+	private _config!: ModuleConfig;
+
+	public init(config: ModuleConfig) {
+		this._config = config;
+	}
+
 	public getKey(address: Buffer, delegateWeight: bigint): Buffer {
 		const buffer = Buffer.alloc(8);
 		buffer.writeBigUInt64BE(delegateWeight);
 		return Buffer.concat([buffer, address]);
 	}
 
 	public async getTop(context: ImmutableStoreGetter, count: number) {
-		return this.iterate(context, { limit: count, reverse: true });
+		return this.iterate(context, {
+			gte: Buffer.alloc(KEY_LENGTH, 0),
+			lte: Buffer.alloc(8 + 20, 255),
+			limit: count,
+			reverse: true,
+		});
 	}
 
 	public async getAll(context: ImmutableStoreGetter) {
-		return this.iterate(context, { reverse: true });
+		return this.iterate(context, {
+			gte: Buffer.alloc(KEY_LENGTH, 0),
+			lte: Buffer.alloc(8 + 20, 255),
+			reverse: true,
+		});
+	}
+
+	public splitKey(key: Buffer): [Buffer, bigint] {
+		const weightBytes = key.slice(0, 8);
+		const address = key.slice(8);
+		return [address, weightBytes.readBigUInt64BE()];
+	}
+
+	public async update(
+		context: StoreGetter,
+		address: Buffer,
+		oldWeight: bigint,
+		delegate: DelegateAccount,
+	): Promise<void> {
+		const oldKey = this.getKey(address, oldWeight);
+		await this.del(context, oldKey);
+
+		const newWeight = getDelegateWeight(
+			BigInt(this._config.factorSelfVotes),
+			delegate.selfVotes,
+			delegate.totalVotesReceived,
+		);
+		if (newWeight < this._config.minWeightStandby) {
+			return;
+		}
+
+		if (delegate.isBanned) {
+			return;
```

Comment:
Move the `delegate.isBanned` check before the new weight calculation, so that if the delegate is banned, there is no need to calculate and compare new weight.